### DISCUSSION TITLE
How to use Theme-Controller component  in Angular?

### DISCUSSION THREAD

**Discussion by Easy-Cloud-in** [saadeghi/daisyui#2521] [Q&A] [answered]
How do I switch themes using Theme-Controller in Angular?

**Comment by saadeghi** [comment, marked as answer]
1. make sure the theme (for example `synthwave` theme) is in your `tailwind.config.js` ([docs](https://daisyui.com/docs/themes/))
```
module.exports = {
  //...
  daisyui: {
    themes: ["light", "dark", "synthwave"],
  },
}
```
2. put this checkbox in your page ([docs](https://daisyui.com/components/theme-controller/))
```
<input type="checkbox" value="synthwave" class="toggle theme-controller"/>
```


Note: `theme-controller` uses CSS `:has()` function so it's not supported in the current version of FireFox as mentioned in the docs:  
<img width="676" alt="css has" src="https://github.com/saadeghi/daisyui/assets/7342023/f145ecda-5a9a-4c4c-8df8-98e6c8984cf3">


Let me know if you have a question.
